### DISCUSSION TITLE
Cannot find module 'next-intl/client' or its corresponding type declarations in the App router example

### DISCUSSION THREAD

**Discussion by BearCooder** [amannn/next-intl#537] [Q&A]
Hello,

when I look at this next-intl app router example here:
https://next-intl-docs-git-feat-next-13-rsc-next-intl.vercel.app/examples/app-router

I want to try it out. But I always get the error:
**Cannot find module 'next-intl/client' or its corresponding type declarations**
when I try to import `import { usePathname, useRouter } from 'next-intl/client';`

I used next-intl latest as it is in the example, I also used the most current beta version "[next-intl](https://npmjs.com/package/next-intl)": "[^3.0.0-beta.19](https://npmjs.com/package/next-intl)", and the next-intl 3.0 release candidate.
Its always the same error? I do not understand why as its the same as in the example.

**Comment by andresilva-cc** [comment]
It seems that the exports for `/client` and `/link` were removed in this commit: https://github.com/amannn/next-intl/commit/549dc178a3415c06e8330a600d9213d63c1c92fb#diff-cf2dbe1e0424ce21f569c7153c4b9df4882a8132a1bff0b0f1d787937155d2ee

@amannn I mean, version 3 adds the support for React Server Components, but I guess we still need those client APIs for some occasions:

1. My menu links are pointing to the version without locale (e.g.: `/about` instead of `/en/about`)

2. I have to use `usePathname` from `next/navigation` and I need to manually strip the locale part of the string to get the active item, instead of just using `usePathname` from `next-intl/client` where the locale is already stripped out.

**Reply by amannn** [reply]
@BearCooder @andresilva-cc Did you have a chance to look at [the release notes](https://next-intl-docs.vercel.app/blog/next-intl-3-0) yet? Your question should be answered there.

**Reply by andresilva-cc** [reply]
Thanks! May I suggest making clearer the difference between the two docs? I mean, I was looking at the [official docs](https://next-intl-docs.vercel.app/) because even though version 3 is still a release candidate, it has a [guide](https://next-intl-docs.vercel.app/docs/getting-started/app-router-server-components) on how to setup with server components using the RC version, but the new `next-intl/navigation` APIs are only available on this other [documentation](https://next-intl-docs-git-feat-next-13-rsc-next-intl.vercel.app/docs/routing/navigation).

**Reply by amannn** [reply]
That's a great point, thanks for bringing this up!

I've added a note at the top of https://next-intl-docs.vercel.app/docs/getting-started/app-router-server-components and reverted the mentioned version below to the latest beta. Hopefully, this should provide some clarification.

If things go well, we might be able to release the new docs in the near future. There were disclaimers and mentions of different versions of `next-intl` in the docs for way too long already 🙂

**Reply by BearCooder** [reply]
Thank you @amannn and @andresilva-cc but I still have one question.

Is the App Router example from next-intl provided in my opening post in this case outdated?
As it says App router but it seems it is not compatible with the recent v3?

**Reply by amannn** [reply]
The example you've referenced above uses the current stable release. You can do that just fine, but it seems like you're looking for Server Components support, in this case [the release candidate](https://next-intl-docs.vercel.app/blog/next-intl-3-0) is what you're looking for.

**Comment by ifeigen** [comment]
So what's the solution?
